controllers/ibu_controller.go

```diff
@@ -142,7 +142,12 @@ func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	if isTransitionRequested(ibu) {
-		if validateStageTransition(ibu) {
+		var isAfterPivot bool
+		isAfterPivot, err = r.RPMOstreeClient.IsStaterootBooted(getStaterootName(ibu.Spec.SeedImageRef.Version))
+		if err != nil {
+			return
+		}
```

Comment:
What do you think moving this check into the `isRollbackAllowed`? Looks like a precondition of that func and reads better instead of passing the bool all the down there?

Reply:
It was like that at first. the problem is `validateStageTransition` and `isRollbackAllowed` don't have access `RPMOStreeClient` or `ImageBasedUpgradeReconciler`. So we can pass this bool or we can pass down the `RPMOstreeClient` or we can make these methods of `ImageBasedUpgradeReconciler`. I thought passing down bool is simpler. Also still open to suggestion.

Reply:
other con of moving this `isRollbackAllowed` is we have to propagate the error back to here. Currently `ValidateStageTransition` and `isRollbackAllowed` don't return any errors.

Reply:
one other con is it makes the unit tests for these two function more complicated, since we have to create a Reconciler to test them, which they don't actually need it.

Reply:
Ah yeah too much refactoring. 

What about adding it in `handleRollback`....which has access to client? Seems like the handlers are meant to check the state of the system (unlike controller funcs...which are really just checking the .status.condtion)....wdyt?

Reply:
if we add it in `handleRollabck` we probably have to go RollbackFailed condition if it is not after pivot.
If we keep it here we go to `InvalidTransition`.
I don't have a preference regarding this.
What do you think @jc-rh ?

Reply:
Gotcha. Totally your call and I wont hold up the PR for sure as this works well! 

Basically I'm just not too sure if makes sense to call `IsStaterootBooted` regardless of which state we are since it's only relevant for rollback.

Reply:
I agree. It is an extra query whenever there is a transition. Although transitions happens two or three times for every upgrade. So time efficiency will not be an issue.
It is a trade off between giving those functions internal access to the `reconciler` and two or three extra queries for every upgrade.
I am still not sure which is better.